pkg/plugins/runtime/k8s/controllers/namespace_controller_test.go

```diff
@@ -78,14 +89,23 @@ var _ = Describe("NamespaceReconciler", func() {
 	})
 
 	It("should delete NetworkAttachmentDefinition when injection annotation is no longer on the namespace", func() {
+		// setup CustomResourceDefinition
+		crd := &v1beta1.CustomResourceDefinition{
+			ObjectMeta: kube_meta.ObjectMeta{
+				Name: "network-attachment-definitions.k8s.cni.cncf.io",
+			},
+		}
+		err := kubeClient.Create(context.Background(), crd)
+		Expect(err).ToNot(HaveOccurred())
+
 		// setup NetworkAttachmentDefinition in the namespace
 		nad := &v1.NetworkAttachmentDefinition{
 			ObjectMeta: kube_meta.ObjectMeta{
 				Namespace: "non-system-ns-without-sidecar-injection",
 				Name:      metadata.KumaCNI,
 			},
 		}
-		err := kubeClient.Create(context.Background(), nad)
+		err = kubeClient.Create(context.Background(), nad)
 		Expect(err).ToNot(HaveOccurred())
 
```

Comment:
should we add a test case for what you changed in this PR?

`It("should skip creating NetworkAttachmentDefinition when CRD is absent in the cluster")`